src/Storages/MergeTree/MergeTreeSequentialSource.cpp

```diff
@@ -142,10 +146,25 @@ MergeTreeSequentialSource::MergeTreeSequentialSource(
         columns_for_reader = data_part->getColumns().addTypes(columns_to_read);
     }
 
-    ReadSettings read_settings;
+    const auto & context = storage.getContext();
+    ReadSettings read_settings = context->getReadSettings();
+    read_settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache = true;
+    /// It does not make sense to use pthread_threadpool for background merges/mutations
+    /// And also to preserve backward compatibility
+    read_settings.local_fs_method = LocalFSReadMethod::pread;
```

Comment:
This change looks unrelated, let's not mix things?

Reply:
This is actually required. Before default ReadSettings was used (which has local_fs_method=pread), now they are used from Context (which has local_fs_method=pread_threadpool). And we do need ReadSettings from context to get throttling.

Reply:
Indeed